### ISSUE TITLE
Proxy array.forEach iterates over unproxied values

### ISSUE THREAD

**Issue by appy-one** [appy-one/acebase-core#4]
When using a live data proxy, forEach on an array value will iterate over the underlying (unproxied) values, which is unexpected behaviour.

Example:
```js
const defaultChatValue = { 
   title: 'Members is an Array with objects',
   description: 'Which is not recommended, but should work anyway',
   members: [{ name: 'Ewout', lastOnline: new Date() }] 
};
const proxy = await db.ref('chats/chat1').proxy(defaultChatValue);
const chat = proxy.value;
chat.members.forEach(member => {
   // member is unproxied by forEach on the members array
   if (member.name === 'Ewout') {
      member.lastOnline = new Date(); // Won't trigger db update!! 😱
   }
});
```

**Comment by appy-one** [comment]
This also applies to array methods like ```every```, ```some```, ```filter```, ```map``` , ```reduce```, ```find``` etc.

Example, ```array.find``` returns unproxied value:
```js
const ewout = chat.members.find(member => member.name === 'ewout');
ewout.lastOnline = new Date(); // Won't trigger update either
```

**Comment by appy-one** [comment]
Released fixes and additions in acebase-core v1.2.0
